### PR TITLE
Add date filters to crab pnl Fixes ENG-803

### PR DESCRIPTION
**Pull request by DemolaJames**
# Task:

## Description

- Add startdate and end date filter to crab strategy's PnL. so users can see pnl for preferred periods
- Add historical premium chart that is on trade page to the crab page

Fixes ENG-803, ENG-812

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

## User Facing Checklist

- [x] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

**Comment by sanandnarayan** [comment]
@DemolaJames can you paste the api PR here?

**Comment by DemolaJames** [comment]
https://github.com/opynfinance/OMDB/pull/3/files

File: packages/frontend/package.json [hunk 1]
```diff
@@ -25,11 +25,14 @@
     "dependencies": {
         "@apollo/client": "^3.4.10",
         "@cypress/react": "^5.12.4",
+        "@date-io/date-fns": "^2.16.0",
+        "@date-io/moment": "1.1.0",
```
Comment (KMKoushik) [inline packages/frontend/package.json:29]: Why moment? Can we just use @date-io/date-fns as we are already using date-fns

File: packages/frontend/src/components/Charts/CrabStrategyV2PnLChart.tsx [hunk 2]
```diff
@@ -2,18 +2,24 @@ import { Box, CircularProgress, Typography} from '@material-ui/core'
...
 import React, { memo, useEffect } from 'react'
 import useAppMemo from '@hooks/useAppMemo'
-import { useCrabPnLV2ChartData } from 'src/state/ethPriceCharts/atoms'
+import { crabv2StrategyFilterEndDateAtom, crabv2StrategyFilterStartDateAtom, useCrabPnLV2ChartData } from 'src/state/crab/atoms'
 import { crabV2graphOptions } from '@constants/diagram'
 import HighchartsReact from 'highcharts-react-official'
 import Highcharts from 'highcharts'
-
+import Grid from "@material-ui/core/Grid";
+import MomentUtils from "@date-io/moment";
```
Comment (KMKoushik) [inline packages/frontend/src/components/Charts/CrabStrategyV2PnLChart.tsx:10]: Can we use date-fns instead? Will it create any issue?

**Review by KMKoushik — APPROVED** [review]

File: packages/frontend/src/components/Charts/CrabStrategyV2PnLChart.tsx (hunk 2 above)
Comment (DemolaJames) [reply packages/frontend/src/components/Charts/CrabStrategyV2PnLChart.tsx]: Changed

**Review by alexisgauba — APPROVED** [review]

**Pull request merged**